Remotion/ObjectBinding/Web.ClientScript/Scripts/BocReferenceValueBase.ts

```diff
@@ -0,0 +1,223 @@
+// but WITHOUT ANY WARRANTY; without even the implied warranty of 
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// 
+
+type ResourceIdentifier = {
+  LoadIconFailedErrorMessage: string,
+  LoadDataFailedErrorMessage: string,
+  SearchStringForDropDownDoesNotMatchRegexMessage: string,
+  NoDataFoundMessage: string,
+  InternetExplorerScreenReaderLabelText: string
+}
+
+type SearchStringValidationInfo = {
+  ValidSearchStringRegex: string,
+  ValidSearchStringForDropDownRegex: string,
+  SearchStringForDropDownDoesNotMatchRegexMessage: string,
+  IgnoreSearchStringForDropDownUponValidInput: string
+}
+
+type CommandInfo = {
+  href: Nullable<string>,
+  target: Nullable<string>,
+  onClick: Nullable<string>,
+  title: Nullable<string>
+}
+
+type IconContext = {
+  businessObjectClass: Nullable<string>,
+  arguments: Nullable<string>
+}
+
+type ErrorHandler = (error: Sys.Net.WebServiceError) => void
+
+type IconInformation = {
+  Url: string,
+  AlternateText: string,
+  ToolTip: string,
+  Width: number,
+  Height: number
+}
+
+class BocReferenceValueBase
+{
+  private static _nullIconUrl: Nullable<string>
+
+  public static InitializeGlobals (nullIconUrl: string): void
+  {
+    BocReferenceValueBase._nullIconUrl = nullIconUrl;
+  };
+
+  public static UpdateCommand (oldCommand: JQuery, businessObject: any, isIconUpdateEnabled: boolean, controlServiceUrl: Nullable<string>, iconContext: Nullable<IconContext>, commandInfo: Nullable<CommandInfo>, onFailure: ErrorHandler): JQuery
+  {
+    ArgumentUtility.CheckNotNull('oldCommand', oldCommand);
+    ArgumentUtility.CheckTypeIsString('businessObject', businessObject);
+    ArgumentUtility.CheckNotNullAndTypeIsBoolean('isIconUpdateEnabled', isIconUpdateEnabled);
+    if (isIconUpdateEnabled)
+      ArgumentUtility.CheckNotNullAndTypeIsString('controlServiceUrl', controlServiceUrl);
+    if (isIconUpdateEnabled)
+      ArgumentUtility.CheckNotNullAndTypeIsObject('iconContext', iconContext);
+    ArgumentUtility.CheckTypeIsObject('commandInfo', commandInfo);
+    ArgumentUtility.CheckTypeIsFunction('onFailure', onFailure);
+
+    var newCommand = BocReferenceValueBase.CreateCommand(oldCommand, commandInfo, businessObject);
+
+    var oldIcon = oldCommand.find('img');
+    var newIcon = BocReferenceValueBase.CreateEmptyIcon(oldIcon, newCommand.prop('title'));
+    newCommand.append(newIcon);
+
+    oldCommand.replaceWith(newCommand);
+
+    if (isIconUpdateEnabled)
+    {
+      var pageRequestManager = Sys.WebForms.PageRequestManager.getInstance();
+      if (pageRequestManager.get_isInAsyncPostBack())
+        return oldCommand;
+
+      var params: any = { businessObject: businessObject };
+      for (var propertyName in iconContext)
```

Comment:
@a-ctor This is a simple property-copy from one type to another. Is there a better way to write this? Should we make a utility for this?

Reply:
`Object.assign(params, iconContext)`

Reply:
@Stefan-Ilic please add a ticket + todo for this.
@a-ctor do we have other places where this is already needed / already a ticket for this?

Reply:
There are some places where `jQuery.extends` is used which is the same functionality. But for this "raw" js usage we do not have an issue yet

Reply:
okay, so those issues get fixed during the jquery migration anyway. we might want to fix this issue together with the query stuff, just because it's the same usecase, but I'm not tied to it.